### PR TITLE
[SYCLomatic] Add support to multi-device for class dpct::device_memory

### PR DESCRIPTION
**Pull request by ziranzha**

### CODE REVIEW

**Review by AndyCHHuang — APPROVED** [review]
LGTM

File: clang/runtime/dpct-rt/include/dpct/memory.hpp [hunk 1]
```diff
@@ -1508,64 +1508,56 @@ class device_memory {
...
     this->~device_memory();
     new (this) device_memory(src, size);
   }
 
   /// Get memory pointer of the memory object, which is virtual pointer when
   /// usm is not used, and device pointer when usm is used.
-  value_t *get_ptr() {
-    return get_ptr(get_default_queue());
-  }
+  value_t *get_ptr() { return get_ptr(get_default_queue()); }
```
Comment (zhimingwang36) [inline clang/runtime/dpct-rt/include/dpct/memory.hpp:1538]: No change to this line,  please don't include code format change.

File: clang/runtime/dpct-rt/include/dpct/memory.hpp [hunk 2]
```diff
@@ -1508,64 +1508,56 @@ class device_memory {
       : device_memory(sycl::range<Dimension>(Arguments...)) {}
 
   ~device_memory() {
-    if (_device_ptr && !_reference)
-      dpct::dpct_free(_device_ptr, _q);
+    if (!_reference) {
```
Comment (zhimingwang36) [inline clang/runtime/dpct-rt/include/dpct/memory.hpp:1511]: Have not found where _reference is set to false?

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**